### PR TITLE
VIH-9045 - Amend Pipeline & Agents Script for SDS

### PR DESCRIPTION
**Pull request by v1ferrarij**
### JIRA link (if applicable) ###

[VIH-9045](https://tools.hmcts.net/jira/browse/VIH-9045)

### Change description ###

- amended Pipeline for SDS
- amended set_up.sh to loop through X amount of agents.
- merged pipeline templates

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: terraform/tags.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+locals {
+  common_tags = module.tags.common_tags
+}
+
+module "tags" {
+  source      = "git::https://github.com/hmcts/terraform-module-common-tags.git?ref=master"
+  environment = "stg"
```
Comment (chris-pateman) [inline terraform/tags.tf:7]: should this be a var, incase we change the subscription? e.g. like testing in dev

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -11,109 +11,126 @@ resources:
...
+
+      - task: ExtractFiles@1
+        displayName: 'Extract Terraform Artifacts'
+        inputs:
+          archiveFilePatterns: '$(Pipeline.Workspace)/$(artifactName)/$(artifactName).zip'
+          destinationFolder: '$(Pipeline.Workspace)/$(artifactName)'
+          cleanDestinationFolder: false
+          overwriteExistingFiles: true
+     
+      - template: ado-pipeline/jobs/terraform-prep.yaml
```
Comment (chris-pateman) [inline azure-pipelines.yaml:121]: Do you not need to reprep the ADO script?

File: ado-pipeline/jobs/prepare-agents-script.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          sed -i 's|REPLACE_AGENTS|${{ parameters.numberOfAgents }}|gI' 'set_up.sh'
+
+  - task: Bash@3
+    displayName: Set DevOps URL
+    inputs:
+      targetType: 'inline'
+      workingDirectory: ${{ parameters.workingDirectory }}
+      script: |
+          echo 'Setting DevOps URL'
+          sed -i 's|REPLACE_DEVOPSURL|${{ parameters.devOpsURL }}|gI' 'set_up.sh'
```
Comment (chris-pateman) [inline ado-pipeline/jobs/prepare-agents-script.yaml:45]: Can these be done in one task?

**Review by chris-pateman — CHANGES_REQUESTED** [review]

File: terraform/tags.tf (hunk 1 above)
Comment (v1ferrarij) [reply terraform/tags.tf:7]: yeah good point. amended.

File: azure-pipelines.yaml [hunk 4]
```diff
@@ -11,109 +11,126 @@ resources:
...
+
+############################################
+# Terraform Apply. #########################
+- stage: DeployAzureAgent
+  displayName: Terraform Apply
+  dependsOn: Wait
+  jobs:
+  - job: Deploy_Terraform
+    displayName: Deploy DevOps Agent
+    steps:
```
[the remark this answers is not in the document]
Comment (v1ferrarij) [reply azure-pipelines.yaml:106]: added - checkout: none

**Review by chris-pateman — APPROVED** [review]
LGTM

**Comment by v1ferrarij** [comment]
Possible Bug.

**Pull request merged**